### PR TITLE
[BPK-2728]: Add 'outline' type to BpkButton

### PR DESCRIPTION
**Pull request by shaundon**
Matches what we already offer for Android and iOS.

### CODE REVIEW

File: packages/bpk-component-button/BpkButtonStory.scss [hunk 1]
```diff
@@ -16,16 +16,12 @@
  * limitations under the License.
  */
 
-/*
- * The purpose of this file is purley to allow the outline button
- * story to have a solid background for viewing and testing. Otherwise
- * it puts the white button on white background
-*/
```
Comment (shaundon) [inline packages/bpk-component-button/BpkButtonStory.scss]: Comment was nice but [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

File: packages/bpk-component-button/BpkButtonStory.scss [hunk 2]
```diff
@@ -16,16 +16,12 @@
...
 @import '~bpk-mixins/index';
 
-.bpk-outline-layout {
-  min-height: 2.625rem;
-  padding: 1.5rem;
+.bpk-button-story-wrapper {
+  padding: $bpk-spacing-xs 0;
+}
 
-  @include bpk-themeable-property(background-color, --bpk-button-featured-background-color, $bpk-button-featured-background-color);
```
Comment (shaundon) [inline packages/bpk-component-button/BpkButtonStory.scss]: Themability was YAGNI - it's nice but it's just extra complexity that isn't required.

File: packages/bpk-component-button/index.js [hunk 3]
```diff
@@ -34,6 +33,5 @@ export {
   secondaryThemeAttributes,
   featuredThemeAttributes,
   destructiveThemeAttributes,
-  outlineThemeAttributes,
```
Comment (shaundon) [inline packages/bpk-component-button/index.js]: Outline buttons don't need to be themed so I removed it

File: packages/bpk-component-button/stories.js [hunk 4]
```diff
@@ -46,364 +46,80 @@ const cssModules = (styles = {}) => className =>
 
 const getClassName = cssModules(STYLES);
 
+const ButtonStory = ({ className, ...rest }: { className: ?string }) => (
```
Comment (shaundon) [inline packages/bpk-component-button/stories.js]: I rewrote most of this file as it had a lot of duplication from way back - it's neater now and removes the need for a separate 'iconOnly' story. It's also now possible to see all buttons on screen together:

Screen Shot 2019-07-22 at 08 17 59

File: packages/bpk-mixins/src/mixins/_buttons.scss [hunk 5]
```diff
@@ -198,28 +198,29 @@
 ///   }
 
 @mixin bpk-button--outline {
-  @include bpk-themeable-property(color, --bpk-button-outline-text-color, $bpk-color-white);
-  @include bpk-themeable-property(background-color, --bpk-button-outline-background-color, transparent);
+  background-color: $bpk-button-outline-background-color;
+  background-image: $bpk-button-outline-background-image;
+  color: $bpk-button-outline-color;
+
+  @include bpk-border-lg($bpk-button-outline-border-color);
```
Comment (shaundon) [inline packages/bpk-mixins/src/mixins/_buttons.scss]: We should *always* define and use specific tokens if possible for mixins, as tokens are the lowest source of truth for the design system. So I've added a load of new ones e.g. `$bpk-button-outline-color`. I mostly copied the ones from the destructive button mixin.

File: packages/bpk-tokens/src/web/base/buttons.json [hunk 6]
```diff
@@ -214,6 +214,58 @@
...
+      "type": "background-image"
+    },
+    "BUTTON_OUTLINE_DISABLED_COLOR": {
+      "value": "{!GRAY_300}",
+      "type": "color"
+    },
+    "BUTTON_OUTLINE_DISABLED_BOX_SHADOW": {
+      "value": "none",
+      "type": "box-shadow"
+    },
```
Comment (shaundon) [inline packages/bpk-tokens/src/web/base/buttons.json]: Aforementioned new tokens.

**Comment by shaundon** [comment]
Design has approved this too - I spoke to Duncan about it.

File: packages/bpk-component-button/README.md [hunk 7]
```diff
@@ -24,6 +24,7 @@ export default () => (
     <BpkButton large>Large primary</BpkButton>
     <BpkButton secondary>Secondary</BpkButton>
     <BpkButton link>Link</BpkButton>
+    <BpkButton outline>Link</BpkButton>
```
Comment (ojcurtis) [inline packages/bpk-component-button/README.md]: Should this be worded Link or rather 'Outline' as the secondary and primary above?

File: packages/bpk-mixins/src/mixins/_buttons.scss [hunk 8]
```diff
@@ -185,6 +185,45 @@
...
 
+/// Outline button. Modifies the bpk-button & bpk-button--secondary mixins.
+///
+/// @require {mixin} bpk-button
+/// @require {mixin} bpk-button--secondary
+///
+/// @example scss
+///   .selector {
+///     @include bpk-button();
+///     @include bpk-button--secondary();
```
Comment (ojcurtis) [inline packages/bpk-mixins/src/mixins/_buttons.scss]: in the `BpkButton.scss` we aren't using the `@include bpk-button--secondary()` mixin so should this comments be updated to reflect this?

File: packages/bpk-component-button/README.md (hunk 7 above)
Comment (shaundon) [reply packages/bpk-component-button/README.md]: Yep, good catch

File: packages/bpk-mixins/src/mixins/_buttons.scss (hunk 8 above)
Comment (shaundon) [reply packages/bpk-mixins/src/mixins/_buttons.scss]: Another good catch :)

**Comment by shaundon** [comment]
@ojcurtis all feedback addressed

**Pull request merged**